File: sentimental-cash/cash.py
from cs50 import get_float


def calculate_quarters(cents):
    """ Returns the number of quarters coins in change"""
    q = 0
    while cents >= 0.25:
        cents = round(cents - 0.25, 2)
        q += 1
    return q


def calculate_dimes(cents):
    """ Returns the number of dime coins in change"""
    d = 0
    while cents >= 0.1:
        cents = round(cents - 0.1, 2)
        d += 1
    return d


def calculate_nickles(cents):
    """ Returns the number of nickle coins in change"""
    n = 0
    while cents >= 0.05:
        cents = round(cents - 0.05, 2)
        n += 1
    return n


def calculate_pennies(cents):
    """ Returns the number of penny coins in change"""
    p = 0
    while cents >= 0.01:
        cents = round(cents - 0.01, 2)
        p += 1
    return p


change = -1
# Prompt user for change owed
while change < 0:
    change = get_float("Change owed: ")

# Calculate quarters
quarters = calculate_quarters(change)
change = round(change - (quarters * 0.25), 2)

# Calculate dimes
dimes = calculate_dimes(change)
change = round(change - (dimes * 0.1), 2)

# Calculate nickels
nickles = calculate_nickles(change)
change = round(change - (nickles * 0.05), 2)

# Calculate pennies
pennies = calculate_pennies(change)
change = round(change - (pennies * 0.01), 2)

# Sum amount of coins
coins = quarters + dimes + nickles + pennies
print(coins)

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // Prompt user for number of cents
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 0);
    return n;
}

int calculate_quarters(int cents)
{
    // Calculate how many quarters
    int q = 0;
    while (cents >= 25)
    {
        cents = cents - 25;
        q++;
    }
    return q;
}

int calculate_dimes(int cents)
{
    // Calculate how many dimes
    int d = 0;
    while (cents >= 10)
    {
        cents = cents - 10;
        d++;
    }
    return d;
}

int calculate_nickels(int cents)
{
    // Calculate how many nickels
    int n = 0;
    while (cents >= 5)
    {
        cents = cents - 5;
        n++;
    }
    return n;
}

int calculate_pennies(int cents)
{
    // Calculate how many pennies
    int p = 0;
    while (cents >= 1)
    {
        cents = cents - 1;
        p++;
    }
    return p;
}
